### PR TITLE
feat: Convert cozy-device-helper to Typescript

### PR DESCRIPTION
**Pull request by acezard**
- Convert codebase to Typescript without modifying runtime execution
- Adds new API to get Flagship App metadata
- Could not maintain platform.js as renamed file

### CODE REVIEW

File: packages/cozy-device-helper/src/index.spec.ts [hunk 1]
```diff
@@ -7,45 +7,60 @@ import {
```
Comment (acezard) [inline packages/cozy-device-helper/src/index.spec.ts:7]: // @tsnocheck en haut du fichier

File: packages/cozy-device-helper/src/flagship.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  navbarHeight?: number
+  platform?: typeof Platform
+  route?: FlagshipRoutes
+  statusBarHeight?: number
+  version?: string
+}
+
+export const flagshipMetadata = window.cozy?.flagship || {}
+
+export const isFlagshipApp = (): boolean => window.cozy?.flagship !== undefined
```
Comment (trollepierre) [inline packages/cozy-device-helper/src/flagship.ts:22]: don't we test this feature?

File: packages/cozy-device-helper/package.json [hunk 3]
```diff
@@ -30,9 +31,12 @@
...
+    "build": "yarn clean && yarn types && babel --extensions .ts,.tsx --ignore '**/*.spec.tsx','**/*.spec.ts' ./src -d ./dist",
+    "clean": "rm -rf ./dist",
     "prepublishOnly": "yarn run build",
-    "test": "jest src/**"
+    "start": "yarn build --watch",
+    "test": "jest src/**",
+    "types": "tsc -p tsconfig-build.json"
   },
-  "types": "index.d.ts"
+  "types": "dist/index.d.ts"
```
Comment (trollepierre) [inline packages/cozy-device-helper/package.json:41]: isn't it a refactor commit?

**Review by trollepierre — APPROVED** [review]

File: packages/cozy-device-helper/src/flagship.ts (hunk 2 above)
Comment (acezard) [reply packages/cozy-device-helper/src/flagship.ts:22]: To be honest, I don't think it's worth it, I'm not even sure what to test except the implementation. For me, TypeScript replaces testing here. It ensures I return a boolean, and it ensures my code will not throw.

File: packages/cozy-device-helper/package.json (hunk 3 above)
Comment (acezard) [reply packages/cozy-device-helper/package.json:41]: yes the location changed after the refactor

Comment (trollepierre) [reply packages/cozy-device-helper/package.json:41]: I think we should rename this commit:
"refactor: Convert cozy-device-helper to Typescript"

File: packages/cozy-device-helper/src/flagship.ts (hunk 2 above)
Comment (trollepierre) [reply packages/cozy-device-helper/src/flagship.ts:22]: i disagree. If a developer does a `window.cozy?.flagship === undefined` by mistakes, nothing will stop the stop, except a good review.

Comment (trollepierre) [reply packages/cozy-device-helper/src/flagship.ts:22]: anyway it's a nit

File: packages/cozy-device-helper/src/flagship.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
+import { Platform } from 'react-native'
```
Comment (Crash--) [inline packages/cozy-device-helper/src/flagship.ts:1]: @acezard We imprt from react-native but I don't see it in devDep. Don't we have webpack errors now ? 

Can't we import from type/react-native or something else?

**Pull request merged**